src/SoCreate.ServiceFabric.PubSub/BrokerService.cs

```diff
@@ -78,7 +79,7 @@ public abstract class BrokerService : StatefulService, IBrokerService
         /// <param name="serviceContext"></param>
         /// <param name="enableAutoDiscovery"></param>
         /// <param name="brokerEventsManager"></param>
-        protected BrokerService(StatefulServiceContext serviceContext, bool enableAutoDiscovery = true, IBrokerEventsManager brokerEventsManager = null)
+        protected BrokerService(StatefulServiceContext serviceContext, bool enableAutoDiscovery = true, IBrokerEventsManager brokerEventsManager = null, IProxyFactories proxyFactories = null)
```

Comment:
We should add the `IProxyFactories` parameter to the other constructor as well so it is always initialized.
Also, I'm thinking it would be nice to have a default implementation of `IProxyFactories` that implements the current implementation instead of checking for if `IProxyFactories` instance is null.